Assets/MirageXR/Common/Scripts/CombinedEditor/ActionDetailView/Annotation Editors/PickAndPlace/PickAndPlaceController.cs

```diff
@@ -104,51 +65,68 @@ public override bool Init(ToggleObject obj)
 
             name = obj.predicate;
 
-            // Set scaling
-            PoiEditor myPoiEditor = transform.parent.gameObject.GetComponent<PoiEditor>();
-            Vector3 defaultScale = new Vector3(0.5f, 0.5f, 0.5f);
-
-            // If everything was ok, return base result.
-            return base.Init(obj);
-        }
-
-        private void LoadPickAndPlacePositions()
-        {
-            var json = File.ReadAllText(Path.Combine(_activityManager.ActivityPath, "pickandplaceinfo/" + _myObj.poi + ".json"));
-
-            Positions positions = JsonUtility.FromJson<Positions>(json);
+            if (!base.Init(obj))
+            {
+                return false;
+            }
 
-            if (_myObj.key == "1")
+            var manipulator = GetComponentInParent<ObjectManipulator>();
+            if (manipulator)
             {
-                _pickObject.localPosition = positions.resetPosition;
+                manipulator.enabled = false;
             }
-            else
+
+            _pickComponent = _pickObject.GetComponent<Pick>();
+            EditModeChanges(_activityManager.EditModeActive);
+
+            if (File.Exists(Path.Combine(_activityManager.ActivityPath, $"pickandplaceinfo/{_myObj.poi}.json")))
             {
-                _pickObject.localPosition = positions.pickObjectPosition;
+                LoadPickAndPlacePositions();
             }
 
+            CheckTrigger();
+
+            return true;
+        }
+
+        private void LoadPickAndPlacePositions()
+        {
+            var json = File.ReadAllText(Path.Combine(_activityManager.ActivityPath, $"pickandplaceinfo/{_myObj.poi}.json"));
+            var positions = JsonUtility.FromJson<Positions>(json);
+
+            _pickObject.localPosition = _myObj.key == "1" ? positions.resetPosition : positions.pickObjectPosition;
             _pickObject.localRotation = positions.pickObjectRotation;
             _pickObject.localScale = positions.pickObjectScale;
+
             _targetObject.localPosition = positions.targetObjectPosition;
             _targetObject.localScale = positions.targetObjectScale != null ? positions.targetObjectScale : _defaultTargetSize;
+
             _pickComponent.MoveMode = positions.moveMode;
             _pickComponent.ResetPosition = positions.resetPosition;
             _pickComponent.ResetRotation = positions.resetRotation;
             _pickComponent.MyModelID = positions.modelID;
 
             if (_pickComponent.MyModelID != string.Empty)
+            {
                 StartCoroutine(LoadMyModel(_pickComponent.MyModelID));
+            }
         }
 
-        public void SavePositions()
+        private void SavePositions()
         {
             if (_myObj == null || _myObj.poi == string.Empty || gameObject == null)
             {
                 return; // only if the poi is instantiated not the prefab
             }
+
+            if (!RootObject.Instance.activityManager.EditModeActive)
```

Comment:
This check should be removed as it prevents changes being saved when in play mode. This means if you move the target or arrow in edit mode, then switch to play mode, and then change step the new position of the objects is not saved.

Reply:
I think this is intended behaviour, isn't it? Because otherwise it would override the edit mode resting position? no?

I think we can simplify the behaviour - and accordingly also remove the reset option (and reset on step change always)?

Reply:
Discussed this with @fwild on slack, looks like the fix would be to save the positions when exiting edit mode.